server/models/singleSourceOfTruthModel.js

```diff
@@ -0,0 +1,40 @@
+const mongoose = require('mongoose');
+
+const {Schema} = mongoose;
+
+const robotMetadataSchema = new Schema({
+    robotId: String,
+    starterId: String
+});
+
+const instructionSchema = new Schema({
+    type: String,
+    name: String,
+    predecessorIds: [mongoose.Types.ObjectId],
+    successorIds: [mongoose.Types.ObjectId],
+    rpaApplication: String,
+    rpaTask: String,
+    rpaParameters: [inputParameterSchema],
+    outputVariable: Schema.Types.Mixed
+});
+
+const inputParameterSchema = new Schema({
+    name: String,
+    value: Schema.Types.Mixed,
+    requireUserInput: Boolean
+});
+
+const markerSchema = new Schema({
+    type: String,
+    name: String,
+    predecessorIds: [mongoose.Types.ObjectId],
+    successorIds: [mongoose.Types.ObjectId]
+});
+
+const SSoT_Schema = new Schema({
+  robotMetadata: robotMetadataSchema,
+  Task: String,
+  Code: String
+}, { collection: 'completeCollection' });
```

Comment:
Probably it is right the way it is but I don't get it. Why does the SSoT_Schema contain Task and Code? Shouldn't it be elements of type array as described in our SSoT_deutsch wiki?

Reply:
This will be changed with the pr fixing #17